rebasehelper/checkers/license_tool.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+#
+# This tool helps you to rebase package to the latest version
+# Copyright (C) 2013-2014 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# he Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Petr Hracek <phracek@redhat.com>
+#          Tomas Hozza <thozza@redhat.com>
+
+from __future__ import print_function
+import os
+from six import StringIO
+
+from rebasehelper.utils import ProcessHelper
+from rebasehelper import settings
+from rebasehelper.checker import BaseChecker
+
+
+class LicenseTool(BaseChecker):
+    """license compare tool"""
+
+    CMD = "licensecheck"
+    DEFAULT = True
+    category = "SOURCE"
+
+    @classmethod
+    def match(cls, cmd=None):
+        if cmd == cls.CMD:
+            return True
+        else:
+            return False
+
+    @classmethod
+    def get_checker_name(cls):
+        return cls.CMD
+
+    @classmethod
+    def is_default(cls):
+        return cls.DEFAULT
+
+    @classmethod
+    def get_license_changes(cls):
+        """
+        Finds out if there is a possibility of license change in new sources
+
+        :return: true if license changes probably happened
+        """
+        old_files = []
+        new_files = []
+        conflicting_files = []
+
+        changed_files = StringIO()
+        old_sources_path = os.path.join(cls.workspace_dir, 'old_sources')
+        new_sources_path = os.path.join(cls.workspace_dir, 'new_sources')
+
+        # Get paths to directories of sources(diff would not handle it, if directory names were different)
+        # Examples of the paths (old_sources/fcitx-qt5-1.1.1, new_sources/fcitx-qt5-1.2.2)
+        old_dir = os.path.join(old_sources_path, os.listdir(old_sources_path)[0])
+        new_dir = os.path.join(new_sources_path, os.listdir(new_sources_path)[0])
+
+        # Example diff output
+        # 'Files fcitx-qt5-1.1.1/guiwrapper/common.h and fcitx-qt5-1.2.2/guiwrapper/common.h differ'
+        ProcessHelper.run_subprocess(["/usr/bin/diff", '-q', '-r', old_dir, new_dir], output_file=changed_files)
```

Comment:
Is calling `diff` really necessary? It's another dependency, and `licensecheck` has `--recursive` option.